services/vehicle/src/main/java/com/wedriveu/services/vehicle/boundary/nearest/VehicleFinderVerticle.java

```diff
@@ -0,0 +1,177 @@
+package com.wedriveu.services.vehicle.boundary.nearest;
+
+import com.wedriveu.services.shared.rabbitmq.RabbitMQConfig;
+import com.wedriveu.services.shared.rabbitmq.VerticleConsumer;
+import com.wedriveu.services.shared.rabbitmq.nearest.VehicleResponse;
+import com.wedriveu.services.shared.utilities.Constants;
+import com.wedriveu.services.shared.utilities.Log;
+import com.wedriveu.services.shared.utilities.Position;
+import com.wedriveu.services.shared.utilities.PositionUtils;
+import com.wedriveu.services.vehicle.entity.Vehicle;
+import com.wedriveu.services.vehicle.rabbitmq.Messages;
+import com.wedriveu.services.vehicle.rabbitmq.UserRequest;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.rabbitmq.RabbitMQClient;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+
+import static com.wedriveu.services.shared.utilities.Constants.*;
+
+/**
+ * Handles the communication with all the current available vehicles.
+ * The available vehicle set has been chosen thanks to the client information data; this Verticle asks for the
+ * actual availability of the vehicles by using RabbitMQ Vert.x library.
+ * At the end of the process, only one vehicle is being picked up.
+ *
+ * @author Marco Baldassarri
+ * @since 5/08/2017
+ */
+public class VehicleFinderVerticle extends VerticleConsumer {
```

Comment:
It's just an advice for you but, these methods (for example `startVehicleCommunication `) are too many lines long, and, since we have a `VehicleConsumer` it seems strange to me that you had to write all that repetitive rabbitmq client code in this class.